pkg/execution/executor/executor.go

```diff
@@ -146,6 +146,13 @@ func WithConfig(c config.Execution) ExecutorOpt {
 	}
 }
 
+func WithFailureHandler(f *func(context.Context, state.Identifier, state.State, state.DriverResponse) error) ExecutorOpt {
```

Comment:
We could type this handler for readability/ease of use:  you can then use this type here and in the struct definition.


```
type FailureHandler func(context.Context, state.Identifier, state.State, state.DriverResponse) error

func WithFailureHandler(f FailureHandler) ExecutorOpt {
```

Reply:
Lovely!